tools/mpy-tool.py

```diff
@@ -1833,6 +1946,124 @@ def extract_segments(compiled_modules, basename, kinds_arg):
                 output.write(source.read(segment.end - segment.start))
 
 
+# derives from compiler-explorer/types/asmresult/asmresult.interfaces.ts
+try:
+    from typing import TYPE_CHECKING
+except ImportError:
+    TYPE_CHECKING = False
+
+if TYPE_CHECKING:
+    from typing import TypedDict, NotRequired
+
+    class AsmResultSource(TypedDict):
+        file: str | None
+        line: int | None
+        column: NotRequired[int]
+        mainsource: NotRequired[bool]
+
+    class AsmResultLink(TypedDict):
+        offset: int
+        length: int
+        to: int
+
+    class AsmResultLabelRange(TypedDict):
+        startCol: int
+        endCol: int
+
+    class AsmResultLabel(TypedDict):
+        name: str
+        target: NotRequired[str]
+        range: AsmResultLabelRange
+
+    class ParsedAsmResultLine(TypedDict):
+        text: str
+        opcodes: NotRequired[list[str]]
+        address: NotRequired[int]
+        disassembly: NotRequired[str]
+        source: NotRequired[AsmResultSource | None]
+        links: NotRequired[list[AsmResultLink]]
+        labels: NotRequired[list[AsmResultLabel]]
+
+    class PartialAsmResultLine(ParsedAsmResultLine):
+        text: NotRequired[str]
+
+    class ParsedAsmResult(TypedDict):
+        asm: list[ParsedAsmResultLine]
+        labelDefinitions: NotRequired[dict[str, int]]
+        parsingTime: NotRequired[int]
+        filteredCount: NotRequired[int]
+        externalParserUsed: NotRequired[bool]
+        objdumpTime: NotRequired[int]
+        execTime: NotRequired[int]
+        languageId: NotRequired[str]
+
+
+global_labels: dict[str, int] = {}
+global_output_line_number: int = 0
+
+
+def line_track_meta_print(*a, meta: PartialAsmResultLine = {}, **k):
```

Comment:
Can this be a `@staticmethod` of `JsonPrinter` to keep everything self contained?